feat(ADR): Documentation Strategy (#129) (#152)
allinbits/cosmos-cash: docs/architecture/adr-002-docs-structure.md

# ADR 002: Documentation Structure

## Status

> PROPOSED

Author: ```suggestion
ACCEPTED
```

- **docs/explanation/adr/README** - introduction to ADR
- **docs/explanation/adr/PROCESS.md** which details how to raise ADRs
- **docs/explanation/adr/adr-template.md** - template for raising ADR
- **docs/explanation/adr/adr-{number}-{desc}.md** - an ADR document

##### docs/Explanation/articles

This SHALL BE a folder that contains a sub-folder for each published article. By published this COULD REFER to blog posts. The folder should be name such that it describes the article purpose. Each sub-folder SHALL CONTAIN all the content relevant to the article (e.g. images, bibliographies etc). These articles can be converted into pdf using pandoc. 

In order to do this

- There SHOULD BE a makefile with targets for calling pandoc - Note: the process for building pdf are not part of the commit or release processes, but ad-hoc
- There SHOULD BE a latex template file that can create pdf files with consistent look and feel. This COULD BE the [Eisvogel template](https://github.com/Wandmalfarbe/pandoc-latex-template) with suitable modifications
- The makefile and template should be independent of the article

> - THere SHOULD BE a README.md that describes how to use and build articles

Reviewer: ```suggestion
- There SHOULD BE a README.md that describes how to use the makefile and template and build articles
```

> > **Note:** Explanations can come in other forms, particularly issue discussion and Pull Requests.

Reviewer: did you mean for that `>` symbol to be here? Not sure what it does

#### docs/Tutorials

> As indicated in the overview, these SHALL BE documents that target beginners and will take a user step-by-step through a process with the aim of achieving some goal. Please see the [Divio tutorial page](https://documentation.divio.com/tutorials/) for detail.

Reviewer: ```suggestion
As indicated in the overview, tutorials SHALL BE documents that target beginners and guide a user step-by-step through a process with the aim of achieving some goal. Please see the [Divio tutorial page](https://documentation.divio.com/tutorials/) for details.
```

> - There SHALL BE a folder for each tutorial - see the [Cosmos SDK tutorials](https://github.com/cosmos/sdk-tutorials) as an example
> - The folder SHALL CONTAIN all the content relevant for that tutorial. 
> - The content SHOULD BE consistent in format with [SDK tutorials](https://tutorials.cosmos.network/). 

Reviewer: ```suggestion
- There SHALL BE a folder for each tutorial. See the [Cosmos SDK tutorials](https://github.com/cosmos/sdk-tutorials) as an example.
- The folder SHALL CONTAIN all of the content that is relevant for that tutorial. 
- The content SHOULD BE consistent in format with [Cosmos SDK tutorials](https://tutorials.cosmos.network/). 
```

#### docs/How-To

> Compared to tutorials, [How-Tos](https://documentation.divio.com/how-to-guides/) are help an experienced reader solve a specific problem. These SHALL USE templates similar to the tutorials - see above.

Reviewer: ```suggestion
In contrast to tutorials, [how-to guides](https://documentation.divio.com/how-to-guides/) are a series of actionable steps to help an experienced reader solve a specific problem. These how-to guides SHALL USE templates similar to the tutorials - see above.
```

### Templates

> Documentation SHOULD USE templates for documentation such as module messages etc. 

Reviewer: ```suggestion
The documentation SHOULD USE Markdown templates to develop structured technical content, like module messages follow templates in the Cosmos SDK. 
```

- [The good docs project](https://github.com/thegooddocsproject)
- [Readme editor](https://readme.so/editor)

#### Code Comments

> These are included as comments also form part of documentation. Comments SHALL FOLLOW recommendation as per [Conventional Comments](https://conventionalcomments.org/)

Reviewer: ```suggestion
PR review comments also form part of the documentation. Comments SHALL FOLLOW recommendation as per [Conventional Comments](https://conventionalcomments.org/)
```

Reviewer: do we mean comments in the PR review process? or what are we talking about here?

```
<label> [decoration]: <subject>

[discussion]
```

where `label = (praise|nitpick|suggestion|issue|question|thought|chore)`

#### Commit Comments

> Commits comments will also follow a similar format as laid out following standard defined by [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary). This SHOULD BE enforced as part of [pre-commit](https://pre-commit.com/) checks

Reviewer: ```suggestion
Commits comments will also follow a similar format as laid out following the standard defined by [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary). This commit convention SHOULD BE enforced as part of [pre-commit](https://pre-commit.com/) checks.
```

## Consequences

This section describes the resulting context, after applying the decision. 

### Backwards Compatibility

> Once this ADR is implemented existing documentation will be migrated from existing sources that include:

Reviewer: ```suggestion
After this ADR is implemented, existing documentation will be migrated from existing sources that include:
```

- Notion
- Other Git repos
- Published papers
- Blog posts 

### Positive

> As a result of this

Reviewer: ```suggestion
As a result of this documentation strategy:
```

> - Content will follow best practice that will be easy to navigate and read

Reviewer: ```suggestion
- Content development and maintenance will follow best practices that ensure content is easy to navigate and read
```

- Content will be in a consistent format

> - Commits, Issues and Pull Requests will follow best practice

Reviewer: ```suggestion
- Commits, Issues, and Pull Requests in the repo will follow best practices
```
